spec/asyncapi.md

```diff
@@ -524,6 +532,42 @@ servers:
 ```
 
 
+#### <a name="serverTraitObject"></a>Server Trait Object
+
+Describes a trait that MAY be applied to a [Server Object](#serverObject). This object MAY contain any property from the [Server Object](#serverObject), except `url`, `protocol` and `protocolVersion`.
```

Comment:
Why not `protocolVersion`? Also, you're missing the `traits` property. Otherwise, traits could define traits that could define traits and so on.

Reply:
Yeah, I missed `traits` 🤦🏼‍♂️ Thanks for catch! However, not specifying `protocolVersion` is deliberate, because it has a correlation with `protocol`. You shouldn't define version like `0.1.0` and then in the server object define `protocol: kafka`, because you don't know if you `kafka` (example) has `0.1.0` version.

We could add `protocolVersion` to the `Server Trait Object`, but together with `protocol`. However `protocol` is mandatory field in the server so we will end up with this same situation as we had with `Operation Object` with `action` and `channel` fields.

Reply:
I think that's completely fine. People usually use one protocol only (two when they use http/ws for frontend too). I think it's up to them to take care that `protocolVersion` still makes sense after applying the trait 🤷

Reply:
Ok, if someone will start to combine with versioning and protocols it will be his/her problem 😄

Reply:
The parser should validate it's a legal combination anyway.